lib/sync/mongodbQueue.js

```diff
@@ -34,10 +34,12 @@ function MongodbQueue(name, metrics, opts) {
   }
   this.queueName = name;
   this.metrics = metrics;
+  this.lock = lock;
+  this.lockName = opts.lockName || ('lock:sync:' + this.queueName);
+  this.lockTimeout = opts.lockTimeout || 10000;
```

Comment:
This lock is only for index creating on the queue collections. It really just to prevent multiple workers to create the indexes at the same time. I think 10 seconds should really be enough as it normally very quick to add indexes. It should have nothing to do with the number of dataset clients